File: app/src/main/java/com/hetica/AutismoCordoba/estadisticasDias.java
package com.hetica.AutismoCordoba;

import androidx.appcompat.app.AppCompatActivity;

import android.app.DatePickerDialog;
import android.content.res.Configuration;
import android.database.Cursor;
import android.os.Bundle;
import android.util.Log;
import android.util.TypedValue;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.DatePicker;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.Calendar;

/**
 * The type Estadisticas dias.
 */
public class estadisticasDias extends AppCompatActivity {

    /**
     * The Db.
     */
    AdminSQLiteOpenHelperStats db;

    /**
     * The Array list.
     */
    ArrayList<String> arrayList;
    /**
     * The Adapter.
     */
    ArrayAdapter<String> adapter;

    /**
     * The Lv.
     */
    ListView lv;

    /**
     * The Year final.
     */
    String yearFinal;
    /**
     * The Month final.
     */
    String monthFinal;
    /**
     * The Day final.
     */
    String dayFinal;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_estadisticas_dias);

        final EditText editText = (EditText) findViewById(R.id.editText2);

        db = new AdminSQLiteOpenHelperStats(this, "Stats.db", null, 1);
        lv = (ListView) findViewById(R.id.listViewDias);

        arrayList = new ArrayList<String>();

        if ((getResources().getConfiguration().screenLayout &
                Configuration.SCREENLAYOUT_SIZE_MASK) ==
                Configuration.SCREENLAYOUT_SIZE_XLARGE) {
            adapter = new ArrayAdapter<String>(estadisticasDias.this, android.R.layout.simple_list_item_1, arrayList){
                @Override
                public View getView(int position, View convertView, ViewGroup parent){
                    /// Get the Item from ListView
                    View view = super.getView(position, convertView, parent);

                    TextView tv = (TextView) view.findViewById(android.R.id.text1);

                    // Set the text size 25 dip for ListView each item
                    tv.setTextSize(TypedValue.COMPLEX_UNIT_DIP,35);

                    // Return the view
                    return view;
                }

            };

        }else
        {
            adapter = new ArrayAdapter<String>(estadisticasDias.this, android.R.layout.simple_list_item_1, arrayList);
        }
        lv.setAdapter(adapter);

        Calendar mcurrentDate=Calendar.getInstance();
        int yearAux = mcurrentDate.get(Calendar.YEAR);
        int monthAux = mcurrentDate.get(Calendar.MONTH);
        int dayAux = mcurrentDate.get(Calendar.DAY_OF_MONTH);
        monthAux=monthAux +1;
        //yearFinal = Integer.toString(monthAux) + Integer.toString(dayAux) + Integer.toString(yearAux);
        if(monthAux<10){
            yearFinal="0"+ Integer.toString(monthAux);
        }else
            {
                yearFinal =Integer.toString(monthAux);
            }

        if(dayAux<10){
            yearFinal=yearFinal + "0";
        }
        yearFinal=yearFinal + Integer.toString(dayAux) + Integer.toString(yearAux);

        editText.setText(dayAux + "/" + monthAux + "/" + yearAux);
        viewData();

        editText.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                // TODO Auto-generated method stub
                //To show current date in the datepicker
                Calendar mcurrentDate=Calendar.getInstance();
                int year = mcurrentDate.get(Calendar.YEAR);
                int month = mcurrentDate.get(Calendar.MONTH);
                int day = mcurrentDate.get(Calendar.DAY_OF_MONTH);
                //month=month +1;
                //yearFinal = Integer.toString(month) + Integer.toString(day) + Integer.toString(year);

                if(month<10){
                    yearFinal="0"+ Integer.toString(month);
                }else
                {
                    yearFinal =Integer.toString(month);
                }
                if(day<10){
                    yearFinal=yearFinal + "0";
                }
                yearFinal=yearFinal + Integer.toString(day) + Integer.toString(year);

                DatePickerDialog mDatePicker=new DatePickerDialog(estadisticasDias.this, new DatePickerDialog.OnDateSetListener() {
                    public void onDateSet(DatePicker datepicker, int selectedYear, int selectedMonth, int selectedDay) {
                        // TODO Auto-generated method stub
                        /*      Your code   to get date and time    */
                        Log.e("Date Selected", "Month: " + selectedMonth + " Day: " + selectedDay + " Year: " + selectedYear);
                        selectedMonth=selectedMonth+1;
                        editText.setText(selectedDay + "/" + selectedMonth + "/" + selectedYear);

                        if(selectedMonth<10){
                            yearFinal="0"+ Integer.toString(selectedMonth);
                        }else
                        {
                            yearFinal =Integer.toString(selectedMonth);
                        }
                        if(selectedDay<10){
                            yearFinal=yearFinal + "0";
                        }
                               yearFinal=yearFinal + Integer.toString(selectedDay) + Integer.toString(selectedYear);
                        viewData();
                    }
                },year, month, day);


                mDatePicker.setTitle("Select date");
                mDatePicker.show();
            }
        });
    }

    /**
     * Función que devuelve las estadísticas de una fecha en concreto
     *
     */
    private void viewData(){

        Log.e("Date Selected", yearFinal);

        Cursor cursor = db.viewDataDias(yearFinal);

        if(cursor.getCount() == 0){
            Toast.makeText(this, "No se trabajó este día", Toast.LENGTH_LONG).show();
            adapter.clear();
            adapter.notifyDataSetChanged();
        }else{
            adapter.clear();
            adapter.notifyDataSetChanged();
            while(cursor.moveToNext()){
                arrayList.add(cursor.getString(1) + "          " + cursor.getString(3) + " minutos");
            }
            if ((getResources().getConfiguration().screenLayout &
                    Configuration.SCREENLAYOUT_SIZE_MASK) ==
                    Configuration.SCREENLAYOUT_SIZE_XLARGE) {
                adapter = new ArrayAdapter<String>(estadisticasDias.this, android.R.layout.simple_list_item_1, arrayList){
                    @Override
                    public View getView(int position, View convertView, ViewGroup parent){
                        /// Get the Item from ListView
                        View view = super.getView(position, convertView, parent);

                        TextView tv = (TextView) view.findViewById(android.R.id.text1);

                        // Set the text size 25 dip for ListView each item
                        tv.setTextSize(TypedValue.COMPLEX_UNIT_DIP,35);

                        // Return the view
                        return view;
                    }

                };

            }else
            {
                adapter = new ArrayAdapter<String>(estadisticasDias.this, android.R.layout.simple_list_item_1, arrayList);
            }
            lv.setAdapter(adapter);
        }

    }
}

File: app/src/main/java/com/hetica/AutismoCordoba/estadisticasDias.kt
package com.hetica.AutismoCordoba

import CustomListAdapter
import CustomToolbarAdapter
import android.annotation.SuppressLint
import android.app.ActivityOptions
import android.app.DatePickerDialog
import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.util.Log
import android.util.TypedValue
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.ImageView
import android.widget.ListView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.navigation.NavigationBarView
import java.util.Calendar



/**
 * The type Estadisticas dias.
 */
class estadisticasDias : AppCompatActivity() {

    var db: AdminSQLiteOpenHelperStats? = null
    var arrayList: ArrayList<String>? = null
    var adapter: CustomListAdapter? = null
    var lv: ListView? = null
    var yearFinal: String? = null
    var monthFinal: String? = null
    var dayFinal: String? = null
    var sumaMinutos: TextView? = null
    var imageMain: ImageView? = null
    var totalMinutos: Double = 0.0
    private lateinit var toolbar: Toolbar

    private lateinit var customToolbarAdapter: CustomToolbarAdapter

    fun setAdminSQLiteOpenHelperStats(db: AdminSQLiteOpenHelperStats) {
        this.db = db
    }

    @SuppressLint("SetTextI18n", "MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_estadisticas_dias)
        val editText = findViewById<View>(R.id.editText2) as EditText
        db = AdminSQLiteOpenHelperStats(this)
        lv = findViewById<View>(R.id.listViewAsignaturas) as ListView
        sumaMinutos = findViewById(R.id.sumaMinutos)
        imageMain=findViewById(R.id.botonMain5)
        GoToMain()

        toolbar = findViewById(R.id.toolbar2)
        setSupportActionBar(toolbar)
        customToolbarAdapter = CustomToolbarAdapter(this, toolbar)
        customToolbarAdapter.setTextSizeBasedOnScreenWidth()

        val bottomNavigation = findViewById<BottomNavigationView>(R.id.bottomNavigationViewestadisticas)
        bottomNavigation.labelVisibilityMode = NavigationBarView.LABEL_VISIBILITY_AUTO
        bottomNavigation.selectedItemId = R.id.action_estadisticas

        bottomNavigation.setOnItemSelectedListener { item ->
            when (item.itemId) {
                R.id.action_estadisticas -> true
                R.id.action_tiempo_dedicado -> {
                    val intent = Intent(this, tiempo_dedicado::class.java)
                    val animationBundle = ActivityOptions.makeCustomAnimation(
                        this,
                        R.anim.slide_in,  // Enter animation
                        R.anim.slide_out  // Exit animation
                    ).toBundle()
                    startActivity(intent, animationBundle)
                    true
                }
                R.id.action_actividad -> {
                    val intent = Intent(this, mayormenoractividad::class.java)
                    val animationBundle = ActivityOptions.makeCustomAnimation(
                        this,
                        R.anim.slide_in,  // Enter animation
                        R.anim.slide_out  // Exit animation
                    ).toBundle()
                    startActivity(intent, animationBundle)
                    true
                }
                R.id.action_comentarios -> {
                    val intent = Intent(this, MostrarComentarios::class.java)
                    val animationBundle = ActivityOptions.makeCustomAnimation(
                        this,
                        R.anim.slide_in,  // Enter animation
                        R.anim.slide_out  // Exit animation
                    ).toBundle()
                    startActivity(intent, animationBundle)
                    true
                }
                R.id.action_calificaciones -> {
                    val intent = Intent(this, VisualizarCalificaciones::class.java)
                    val animationBundle = ActivityOptions.makeCustomAnimation(
                        this,
                        R.anim.slide_in,  // Enter animation
                        R.anim.slide_out  // Exit animation
                    ).toBundle()
                    startActivity(intent, animationBundle)
                    true
                }
                else -> false
            }
        }

        arrayList = ArrayList()
        adapter = CustomListAdapter(this, android.R.layout.simple_list_item_1, arrayList!!)
        lv!!.adapter = adapter

        val mcurrentDate = Calendar.getInstance()
        val yearAux = mcurrentDate[Calendar.YEAR]
        var monthAux = mcurrentDate[Calendar.MONTH]
        val dayAux = mcurrentDate[Calendar.DAY_OF_MONTH]
        monthAux = monthAux + 1

        yearFinal = if (monthAux < 10) {
            "0" + Integer.toString(monthAux)
        } else {
            Integer.toString(monthAux)
        }
        if (dayAux < 10) {
            yearFinal = yearFinal + "0"
        }
        yearFinal = yearFinal + Integer.toString(dayAux) + Integer.toString(yearAux)
        editText.setText("$dayAux/$monthAux/$yearAux")
        viewData()
        editText.setOnClickListener {
            val mcurrentDate2 = Calendar.getInstance()
            val year = mcurrentDate2[Calendar.YEAR]
            val month = mcurrentDate2[Calendar.MONTH]
            val day = mcurrentDate2[Calendar.DAY_OF_MONTH]
            yearFinal = if (month < 10) {
                "0" + Integer.toString(month)
            } else {
                Integer.toString(month)
            }
            if (day < 10) {
                yearFinal = yearFinal + "0"
            }
            yearFinal = yearFinal + Integer.toString(day) + Integer.toString(year)
            val mDatePicker = DatePickerDialog(this@estadisticasDias, { _, selectedYear, selectedMonth, selectedDay ->
                var adjustedMonth = selectedMonth
                Log.e("Date Selected", "Month: $adjustedMonth Day: $selectedDay Year: $selectedYear")
                adjustedMonth = adjustedMonth + 1
                editText.setText("$selectedDay/$adjustedMonth/$selectedYear")
                yearFinal = if (adjustedMonth < 10) {
                    "0" + Integer.toString(adjustedMonth)
                } else {
                    Integer.toString(adjustedMonth)
                }
                if (selectedDay < 10) {
                    yearFinal = yearFinal + "0"
                }
                yearFinal = yearFinal + Integer.toString(selectedDay) + Integer.toString(selectedYear)
                viewData()
            }, year, month, day)
            mDatePicker.setTitle("Select date")
            mDatePicker.show()
        }
    }

    fun GoToMain(){
        imageMain?.setOnClickListener {
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }
    }

    @SuppressLint("Range")
    private fun viewData() {
        Log.e("Date Selected", yearFinal!!)
        val day = yearFinal!!.substring(0, 2)
        val month = yearFinal!!.substring(2, 4)
        val year = yearFinal!!.substring(4)
        val formattedDate = "$month/$day/$year"
        val cursor = db!!.viewDataDias(formattedDate)
        Log.e("Date Formatted", formattedDate)
        var nodatos: TextView?=findViewById(R.id.textoNoComentarios4)
        if (cursor.count == 0) {
            if (nodatos != null) {
                nodatos.visibility=View.VISIBLE
            }
            adapter!!.clear()
            adapter!!.notifyDataSetChanged()
        } else {
            if (nodatos != null) {
                nodatos.visibility=View.INVISIBLE
            }
            adapter!!.clear()
            adapter!!.notifyDataSetChanged()
            while (cursor.moveToNext()) {
                val asignatura = cursor.getString(1)
                val minutos = cursor.getString(3)
                Log.d("Datos de la base de datos", "Asignatura: $asignatura, Minutos: $minutos")
                arrayList!!.add("$asignatura          $minutos minutos")
                totalMinutos += minutos.toDouble()
            }

            sumaMinutos?.text = "Total: $totalMinutos minutos"
            totalMinutos=0.0
            lv!!.adapter = adapter
        }

    }
}
